### PR TITLE
refactor: migration mongodb to supabase for recent users

### PR DESCRIPTION
**Pull request by dhairyathedev**
- Updated the API calls
- Removed mongoose from package.json
- Updated Documentation

Please find the updated documentation for creation of table in supabase.

### CODE REVIEW

**Comment by dhairyathedev** [comment]
@ashutosh-rath02 Please find the step 7 in the documentation to create table in your supabase database. https://github.com/dhairyathedev/git-re/tree/main?tab=readme-ov-file#setup-and-installation

**Comment by ashutosh-rath02** [comment]
Hey @dhairyathedev just a suggestion do test this properly before pushing as we will be shifting the db. Also the README is getting packed, I think we should transfer the setup part to a new file Learn.md and mention that in README

File: app/api/users/route.ts [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
+      .from("recent_users")
+      .select("*")
+      .eq("username", git_username)
+      .single();
 
-    if (existingUser) {
-      console.log("user already exits");
-      return new Response("Username updated successfully", { status: 200 });
+    if (existingUser.data && existingUser.data.name) {
+      console.log("User already exists");
```
Comment (ashutosh-rath02) [inline app/api/users/route.ts:22]: remove consoles and display data in reverse order as we need recent user first

**Review by ashutosh-rath02 — CHANGES_REQUESTED** [review]
Remove all console logs, instead push status with status codes

File: app/api/users/route.ts (hunk 1 above)
Comment (dhairyathedev) [reply app/api/users/route.ts:22]: > remove consoles and display data in reverse order as we need recent user first

Hey @ashutosh-rath02, the data is displayed in the reverse order, meansing new user at first.
```js
// /app/api/users/route.ts 
const { data: users, error } = await supabase
      .from("recent_users")
      .select("*")
      .order("id", { ascending: false })
      .limit(7);
```
we have kept `ascending: false` which returns the latest user first.

**Comment by dhairyathedev** [comment]
> Hey @dhairyathedev just a suggestion do test this properly before pushing as we will be shifting the db. Also the README is getting packed, I think we should transfer the setup part to a new file Learn.md and mention that in README

I have tested and it works completely fine. Yes, we can move Supabase Authentication and Table setup to learn.md. 

Will you be able to create learn.md docs?

@ashutosh-rath02

**Comment by ashutosh-rath02** [comment]
> > Hey @dhairyathedev just a suggestion do test this properly before pushing as we will be shifting the db. Also the README is getting packed, I think we should transfer the setup part to a new file Learn.md and mention that in README
> 
> I have tested and it works completely fine. Yes, we can move Supabase Authentication and Table setup to learn.md.
> 
> Will you be able to create learn.md docs?
> 
> @ashutosh-rath02

You can create a Learn.md and push all setup files there

**Comment by dhairyathedev** [comment]
> > > Hey @dhairyathedev just a suggestion do test this properly before pushing as we will be shifting the db. Also the README is getting packed, I think we should transfer the setup part to a new file Learn.md and mention that in README
> > 
> > 
> > I have tested and it works completely fine. Yes, we can move Supabase Authentication and Table setup to learn.md.
> > Will you be able to create learn.md docs?
> > @ashutosh-rath02
> 
> You can create a Learn.md and push all setup files there

Done @ashutosh-rath02

**Comment by ashutosh-rath02** [comment]
@all-contributors please add @dhairyathedev for code maintenance

**Pull request merged**